### ISSUE TITLE
TODO: Mana pool calculation rework

### ISSUE THREAD

**Issue by GizzleBizzle** [The-Alpha-Project/alpha-core#1319]
Similarly to health (as I wrote in #1147), different classes also got different amounts of mana per a point of intellect. It wasn't flat 15 mana for everybody. Here's how it should look like:

The calculation is very similar to the health one. The first 20 points of intellect grant 40 (so 2 mana per 1 int) mana for mana-using classes, however every point after the first 20 uses the following calculation:

Paladins should get 16 mana per int
Shamans - 18
Druids - 18*
Mages - 20
Priests - 20
Warlocks - 20

* druids were introduced in 0.6, where a different calculation was used (Blizzard changed their hp calculation from 1 sta = 16hp to 1 sta = 7 hp), but it seems that mana calculation stayed the same. If not, it still fits quite nicely with other hybrid classes getting 18 mana per int.

**Source**
My own mental sudoku google doc, where I use screenshots as proof to find relations between classes and how many attributes/hpoints and mana points they get each lvl (for another report. Our hp/mana on lvling up is totally off the charts)
https://docs.google.com/spreadsheets/d/1tOgZ4qSakVoV8CrFGI1pNLnjUh-D5BGqMvx3z9y9DuA/edit?usp=sharing

**Comment by GrenderG** [comment]
Hmmm at this moment the first 20 points of intellect are granting 20 points of mana (like in Vanilla). Where did you find that they should give 40 points of mana in total?

**Comment by GizzleBizzle** [comment]
Well, first of all we have got the wrong starting attributes for all classes because we are most likely using the 1.12.1 ones.
I've got almost all of them already researched. I will make a report once I get the last ones.


Let's analyze some examples, because it isn't a single screenshot with someone talking about it.
All screenshots below are from 0.5.5 or 0.6 when it comes to druids (cause they became playable in this patch) or 0.5.4 when it comes to shamans because we got lucky and we've got actual 100% correct data for them.

**Human mages**
https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Eastern%20Kingdoms/Elwynn%20Forest/0320wow05.jpg

Human mages (should) start with 23 stamina, 3 points of intellect over base 20 and a mana pool of 100. If we assume one point of intellect grants 20 mana to mages, we can subtract the amount of mana gained from intellect alone to get our base mana pool.
In this case (3x20) -100 = 40 base mana pool

**Human Paladins**
https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Eastern%20Kingdoms/Elwynn%20Forest/screenhot_betatest_quest3.jpg

Human paladins (should) start with 21 intellect, one point above base and 56 mana. If we assume that one point of intellect gives paladins 16 mana we get (1x16) - 56 = 40 base mana pool

**Gnome warlocks**
https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Eastern%20Kingdoms/Dun%20Morogh/world-of-warcraft-ss101091530907043_1083106.jpg
(Should) start with 24 int and 120 mana. (4x20) - 100 = 40 base mana pool

**Night Elf Druids**
https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Kalimdor/Teldrassil/jokey_WoWScrnShot_041704_193711_resize.jpg

(Should) start with 23 intellect, 3 over base with 94 mana. (3x18) - 100 = 40... always 40....

That would mean, that the first 20 points of intellect give 2 mana per point.
